torch/onnx/_internal/fx/type_utils.py

```diff
@@ -81,6 +82,14 @@ def from_torch_dtype_to_abbr(dtype: Optional[torch.dtype]) -> str:
     return _TORCH_DTYPE_TO_ABBREVIATION.get(dtype, "")
 
 
+def from_scalar_type_to_torch_dtype(scalar_type: type) -> torch.dtype:
+    return _SCALAR_TYPE_TO_TORCH_DTYPE[scalar_type]
+
+
+def maybe_from_scalar_type_to_torch_dtype(scalar_type: type) -> Optional[torch.dtype]:
+    return _SCALAR_TYPE_TO_TORCH_DTYPE.get(scalar_type)
```

Comment:
I also had this issue. Sometimes we are certain it's going to return and sometimes not. DO you think having two functions are the most efficient way to deal with this situation?

I wonder if we should keep one, and have the "check None" logic in where it calls the function.

Reply:
We should unify the rule before we all start coding in our style I think.
@justinchuby thoughts?

Reply:
I feel there should be a better way. But I don’t have an idea yet.

Reply:
+1 Titai's point. Duplicating code with corner case is not the best option. Having the `allow_none=False` or something like that is more sensible

Reply:
> I also had this issue. Sometimes we are certain it's going to return and sometimes not. DO you think having two functions are the most efficient way to deal with this situation?
> 
> I wonder if we should keep one, and have the "check None" logic in where it calls the function.

I think it depends on which branch is used more often. If most of the cases the return is not None, it is unnecessarily verbose putting the check None logic in every call site. That being said, I'm changing code to merge the two since both are used only at one place for now.

Reply:
> +1 Titai's point. Duplicating code with corner case is not the best option. Having the allow_none=False or something like that is more sensible

I don't like this idea. The return type will be annotated with `Optional`, mixing the two scenarios for type system. One will always need to perform not None assertion to satisfy mypy even if `allow_none=False` is set.

> Might as well just use dict.get() because this mapping is internal to torch.onnx.

Seems against the current style that dicts are not accessed directly outside of the file 🤷.